src/Sarif.Multitool/FileWorkItemsCommand.cs

```diff
@@ -76,18 +77,19 @@ public int Run(FileWorkItemsOptions options, IFileSystem fileSystem)
                         logsToProcess = visitor.SplitSarifLogs;
                     }
 
-                    IList<WorkItemFilingMetadata> workItemMetadata = new List<WorkItemFilingMetadata>(logsToProcess.Count);
+                    IList<WorkItemModel> workItemModels = new List<WorkItemModel>(logsToProcess.Count);
 
                     for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
                     {
                         SarifLog splitLog = logsToProcess[splitFileIndex];
-                        workItemMetadata.Add(splitLog.CreateWorkItemFilingMetadata(projectName, options.TemplateFilePath, options.GroupingStrategy));
+                        WorkItemModel workItemModel = splitLog.CreateWorkItemModel(projectName);
+                        workItemModels.Add(workItemModel);
                     }
 
                     try
                     {
-                        IEnumerable<WorkItemFilingMetadata> filedWorkItems = filer.FileWorkItems(options.ProjectUri, workItemMetadata, options.PersonalAccessToken).Result;
-
+                        string securityToken = Environment.GetEnvironmentVariable("SarifWorkItemFilingSecurityToken");
+                        IEnumerable<WorkItemModel> filedWorkItems = filer.FileWorkItems(options.ProjectUri, workItemModels, securityToken).Result;
```

Comment:
>Result [](start = 139, length = 6)

Minor: Is it possible if we don't block here?

Reply:
Can you clarify your comment? :)

---
In reply to: [375953433](https://github.com/microsoft/sarif-sdk/pull/1744#discussion_r375953433) [](ancestors = 375953433)

Reply:
This is probably outside of the scope of this PR. Since this is an async method, ideally it should await but then we would have to update Run() to return Task<int> instead of int. I wouldn't worry too much about this right but just something for future thoughts. 

---
In reply to: [375957765](https://github.com/microsoft/sarif-sdk/pull/1744#discussion_r375957765) [](ancestors = 375957765,375953433)

Reply:
Got it. The way this thing works currently is that we must generate a new SARIF file that contains the URLs to the constructed work items. There's no option currently to skip this (i.e., to simply fire the work item creation request and not wait around for it). That option could be important in the future, I agree. 

---
In reply to: [375968167](https://github.com/microsoft/sarif-sdk/pull/1744#discussion_r375968167) [](ancestors = 375968167,375957765,375953433)